core/include/detray/propagator/navigation_policies.hpp

```diff
@@ -0,0 +1,94 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// detray definitions
+#include "detray/definitions/qualifiers.hpp"
+#include "detray/propagator/base_actor.hpp"
+
+// system includes
+#include <climits>
+
+namespace detray {
+
+/// Struct that represents the most conservative navigation policy: alway re-
+/// initialize the current volume
+struct always_init : actor {
+
+    struct state_type {};
+
+    /// Sets the navigation trust level to 'no trust'
+    ///
+    /// @param pol_state not used
+    /// @param propagation state of the propagation
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE inline void operator()(
+        const state_type & /*pol_state*/,
+        propagator_state_t &propagation) const {
+        propagation._navigation.set_no_trust();
+    }
+};
+
+/// During guided navigation only the next surface should be re-evaluated. This
+/// maps to the 'high trust' level in the navigator
+struct guided_navigation : actor {
+
+    struct state_type {};
+
+    /// Sets the navigation trust level to 'no trust'
+    ///
+    /// @param pol_state not used
+    /// @param propagation state of the propagation
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE inline void operator()(
+        const state_type & /*pol_state*/,
+        propagator_state_t &propagation) const {
+        propagation._navigation.set_high_trust();
+    }
+};
+
+namespace step {
```

Comment:
hmm,, but why?

Reply:
Other actors might also define policies at some point

Reply:
Then I would change the struct name to `default_navigation_policy`. Adding namespace here feels a bit odd to me

Reply:
Renamed